torch/csrc/autograd/variable.cpp

```diff
@@ -179,6 +179,7 @@ void Variable::Impl::set_data(Tensor new_data) {
 }
 
 void Variable::Impl::release_resources() {
+  data_.unsafeGetTensorImpl()->detach_autograd_meta();
```

Comment:
Just discussed with @ezyang : this is necessary only because Variable::Impl is the actual owner of data_'s `autograd_meta_`. I should store the autograd metadata in the Variable::Impl's `autograd_meta_` instead, and release it appropriately here.